lib/lroPollStrategy.ts

```diff
@@ -102,7 +105,15 @@ export abstract class LROPollStrategy {
       this._status = getProvisioningState(result.parsedBody) || "Succeeded";
       this._response = result;
       this._mostRecentRequest = result.request;
-      this._resource = getResponseBody(result);
+      try {
+        this._resource = getResponseBody(result);
+      } catch (error) {
+        this._resource = undefined;
+        const resultStatus: number = result.status;
+        if (this._initialRequestMethod !== "DELETE" || resultStatus < 400 || 499 < resultStatus) {
```

Comment:
So is the idea that getResponseBody is allowed to fail, but only when either:

- the original request method was a DELETE
- the response status (for the original request? the most recent request?) indicates a request error

Reply:
Almost. It's not an either (or), it's an and. If the initial request was a DELETE and the result status is in the 400s, then we throw.

Reply:
I'm struggling to reconcile the code with your description just by staring at them, but if the tests pass I'm happy. This code makes it seem like an initial `"GET"` or a `"POST"` will always result in a rethrow, for instance.

Reply:
Oh, I'm sorry. I'm an idiot. You're right. If it's a delete and in the 400s, then we don't want to throw. All other scenarios that produce an error while parsing the response body do want to throw. You were right.